olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -97,6 +97,16 @@ client::CancellationToken VersionedLayerClientImpl::GetPartitions(
                    std::move(partitions_task), std::move(callback));
   };
 
+  if (request.GetFetchOption() == CacheWithUpdate) {
```

Comment:
Why don't you move this up top? Why extra create the schedule_get_partitions closure just to not execute it?
Same for below methods.